hazelcast/src/test/java/com/hazelcast/jet/core/JobTest.java

```diff
@@ -787,6 +791,63 @@ public void when_suspendedNamedJob_then_newJobWithEqualNameFails() {
                 .isInstanceOf(JobAlreadyExistsException.class);
     }
 
+    @Test
+    public void given_suspensionIsForbidden_when_suspendJob_then_jobIsTerminated() {
+        // Given
+        DAG dag = new DAG().vertex(new Vertex("test", new MockPS(NoOutputSourceP::new, NODE_COUNT * 2)));
+        JobConfig jobConfig = new JobConfig();
+        jobConfig.setArgument(JobConfigArguments.KEY_JOB_IS_SUSPENDABLE, false);
+
+        // When
+        Job job = instance().getJet().newJob(dag, jobConfig);
+
+        // Then
+        assertJobStatusEventually(job, RUNNING);
+        assertThatThrownBy(job::suspend)
+                .hasMessageContaining("Cannot suspend the job being analyzed");
+
+        cancelAndJoin(job);
+    }
+
+    @Test
+    public void given_suspensionIsForbidden_when_restartJob_then_jobIsTerminated() {
+        // Given
+        DAG dag = new DAG().vertex(new Vertex("test", new MockPS(NoOutputSourceP::new, NODE_COUNT * 2)));
+        JobConfig jobConfig = new JobConfig();
+        jobConfig.setArgument(JobConfigArguments.KEY_JOB_IS_SUSPENDABLE, false);
+
+        // When
+        Job job = instance().getJet().newJob(dag, jobConfig);
+
+        // Then
+        assertJobStatusEventually(job, RUNNING);
+        assertThatThrownBy(job::restart)
+                .hasMessageContaining("Cannot suspend the job being analyzed");
+
+        cancelAndJoin(job);
+    }
+
+    @Test
+    public void given_suspensionIsForbidden_when_changedClusterState_then_jobIsCanceled() {
+        // Given
+        DAG streamingDag = new DAG();
+        streamingDag.newVertex("v", () -> new MockP().streaming());
+        JobConfig jobConfig = new JobConfig().setName("foo").setArgument(KEY_JOB_IS_SUSPENDABLE, false);
+
+        // When
+        Job activeJob = instance().getJet().newJob(streamingDag, jobConfig);
+
+        // Then
+        assertJobStatusEventually(activeJob, RUNNING);
+        instance().getCluster().changeClusterState(ClusterState.PASSIVE);
+        instance().getCluster().changeClusterState(ClusterState.ACTIVE);
+        assertJobStatusEventually(activeJob, FAILED);
+
+        JobResult jobResult = getJetServiceBackend(instance()).getJobRepository().getJobResult(activeJob.getId());
```

Comment:
simpler to use `Job` API

Reply:
Hmm, don't see which one

Reply:
https://github.com/hazelcast/hazelcast/pull/25939#discussion_r1390970222